apache2/apache2_config.c

```diff
@@ -1557,18 +1547,23 @@ static const char *cmd_data_dir(cmd_parms *cmd, void *_dcfg, const char *p1)
 #ifdef WAF_JSON_LOGGING_ENABLE
     strcpy( wafjsonlog_path, dcfg->data_dir );
     strcat( wafjsonlog_path, WAF_LOG_UTIL_FILE );
-    rc = apr_file_open(&dcfg->wafjsonlog_fd, wafjsonlog_path,
+    rc = apr_file_open(&msc_waf_log_fd, wafjsonlog_path,
                    APR_WRITE | APR_APPEND | APR_CREATE | APR_BINARY,
                    CREATEMODE | APR_WREAD, cmd->pool);
 
     if (rc != APR_SUCCESS) {
         return apr_psprintf(cmd->pool, "ModSecurity: Failed to open wafjson log file: %s",
             wafjsonlog_path);
     }
+    
+    // Global variable to share between threads
+    strcpy( msc_waf_log_path, wafjsonlog_path );
```

Comment:
Using `strcpy()` is highly discouraged as this function is dangerous. It can easily overrun the destination buffer if it is not big enough and clobber bytes outside it causing memory corruption. 

Please either use `strncpy()` which is slightly better or look up if Apache API already provides some helper for safe string copying - I would expect there shoul be one.

Reply:
changed